src/index.ts

```diff
@@ -92,6 +100,36 @@ export function euclid(numberOfTrue: number, numberOfFalse: number): boolean[] {
   return bjorklund(subsequences).reduce((a, b) => a.concat(b), []);
 }
 
+// This algorithm, a variant of the Bresenham line algorithm, returns the "brightest mode" of
+// the "scale" where `first` is treated as larger than `second`.
+// It's based on following the closest approximation of the line y = b/a*x that is strictly below the line.
+function bresenham(a: number, b: number, first: any, second: any): any[] {
+  const d = gcd(a, b);
+  if (d === 1) {
+    const result: boolean[] = [];
+    // `xHere` = current number of `first`, `yHere` = current number of `second`; start at (0, 0).
+    let [xHere, yHere] = [0, 0];
+    while (xHere < a || yHere < b) {
```

Comment:
`while` loops can be tricky. Remember to add tests for the cases where `a` or `b` is zero, negative or `NaN` in case of user mis-input. Don't want to hang the app.

Reply:
I guess `a` and `b` should be positive non-zero integers based on the context. This whole package could use some input validation... That's on me.